lms/djangoapps/badges/backends/badgr.py

```diff
@@ -0,0 +1,171 @@
+"""
+Badge Awarding backend for Badgr-Server.
+"""
+import hashlib
+import logging
+import mimetypes
+
+import requests
+from django.conf import settings
+from lazy import lazy
+from requests.packages.urllib3.exceptions import HTTPError
+
+from badges.backends.base import BadgeBackend
+from eventtracking import tracker
+
+from badges.models import BadgeAssertion
+
+MAX_SLUG_LENGTH = 255
+LOGGER = logging.getLogger(__name__)
+
+
+class BadgrBackend(BadgeBackend):
+    """
+    Backend for Badgr-Server by Concentric Sky. http://info.badgr.io/
+    """
+    badges = []
+
+    def __init__(self):
+        assert settings.BADGR_API_TOKEN
+
+    @lazy
+    def _base_url(self):
+        """
+        Base URL for all API requests.
+        """
+        return "{}/v1/issuer/issuers/{}".format(settings.BADGR_BASE_URL, settings.BADGR_ISSUER_SLUG)
+
+    @lazy
+    def _badge_create_url(self):
+        """
+        URL for generating a new Badge specification
+        """
+        return "{}/badges".format(self._base_url)
+
+    def _badge_url(self, slug):
+        """
+        Get the URL for a course's badge in a given mode.
+        """
+        return "{}/{}".format(self._badge_create_url, slug)
+
+    def _assertion_url(self, slug):
+        """
+        URL for generating a new assertion.
+        """
+        return "{}/assertions".format(self._badge_url(slug))
+
+    def _slugify(self, badge_class):
+        """
+        Get a compatible badge slug from the specification.
+        """
+        slug = badge_class.issuing_component + badge_class.slug
+        if badge_class.issuing_component and badge_class.course_id:
```

Comment:
This is for compatibility with the old MVP implementation. Those badges will not have their issuing component field set. All new ones should.